src/components/select/Select.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+import style from './select.module.scss';
+import { Option } from '../../types';
+const { select, selectOptions, optionLanguage, selectContainer } = style;
+
+interface selectProps {
+  placeholder?: string;
+  options: Option[];
+  onChange: (selection: Option) => void;
+  selectedValue: Option | null;
+}
+
+const Select = ({
+  placeholder,
+  options,
+  onChange,
+  selectedValue,
+}: selectProps) => {
+  const [showOptions, setShowOptions] = useState<boolean>(false);
+
+  return (
+    <div className={selectContainer}>
+      <div onClick={() => setShowOptions(!showOptions)} className={select}>
+        <span>{selectedValue ? selectedValue.label : placeholder}</span>
+        <span>v</span>
+      </div>
+      {showOptions && (
+        <div className={selectOptions}>
+          {options.map((option) => (
+            <div
+              onClick={() => {
+                onChange(option);
+                setShowOptions(false);
+              }}
```

Comment:
do not call more than 1 function inside a callback defined directly inside of an onClick statement, some [doc](https://upmostly.com/tutorials/adding-multiple-functions-inside-a-single-onclick-event-handler) here